include/openenclave/enclave.h

```diff
@@ -724,6 +724,78 @@ oe_enclave_t* oe_get_enclave(void);
  */
 oe_result_t oe_random(void* data, size_t size);
 
+/**
+ * oe_generate_attestation_certificate.
+ *
+ * This function generates a self-signed x.509 certificate with an embedded
+ * quote from the underlying enclave.
+ *
+ * @param[in] subject_name a string contains an X.509 distinguished
+ * name (DN) for customizing the generated certificate. This name is also used
+ * as the issuer name because this is a self-signed certificate
+ * See RFC5280 (https://tools.ietf.org/html/rfc5280) specification for details
+ * Example value "CN=Open Enclave SDK,O=OESDK TLS,C=US"
+ *
+ * @param[in] private_key a private key used to sign this certificate
+ * @param[in] private_key_size The size of the private_key buffer
+ * @param[in] public_key a public key used as the certificate's subject key
+ * @param[in] public_key_size The size of the public_key buffer.
+ *
+ * @param[out] output_cert a pointer to buffer pointer
+ * @param[out] output_cert_size size of the buffer above
+ *
+ * @return OE_OK on success
+ */
+oe_result_t oe_generate_attestation_certificate(
+    const unsigned char* subject_name,
+    uint8_t* private_key,
+    size_t private_key_size,
+    uint8_t* public_key,
+    size_t public_key_size,
+    uint8_t** output_cert,
+    size_t* output_cert_size);
+
+/**
+ * Free the given cert
+ * @param[in] cert If not NULL, the buffer to free.
+ */
+void oe_free_attestation_certificate(uint8_t* cert);
```

Comment:
style comment: In Microsoft public headers and sample code, we try to avoid abbreviations not just in method names, but also variable/parameter names.  "cert" should be expanded to "certificate" in 755, 756, 760, 762, etc., for consistency with the method name.  (Acronyms on the other hand are fine, so "der" is perfectly fine in 794.)   There are a very small number of exceptions where abbreviations are ok.  Those include "app" and "arg" (which is used in this file and is ok.)